### PR TITLE
hugo: 0.32.2 -> 0.36.1

### PR DESCRIPTION
**Pull request by monotux**
###### Motivation for this change
Update to a more recent version. Cherry picked from master as I know little to nothing on how to build go packages.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ryantm** [comment]
@GrahamcOfBorg build hugo

File: pkgs/applications/misc/hugo/deps.nix [hunk 1]
```diff
@@ -72,16 +72,7 @@
...
-    fetch = {
-      type = "git";
-      url = "https://github.com/dchest/cssmin";
-      rev = "fb8d9b44afdc258bfff6052d3667521babcb2239";
-      sha256 = "09sdijfx5d05z4cd5k6lhl7k3kbpdf2amzlngv15h5v0fff9qw4s";
-    };
-  }
-  {
-    goPackagePath = "github.com/dlclark/regexp2";
+    goPackagePath = "github.com/disintegration/imaging";
```
Comment (ryantm) [inline pkgs/applications/misc/hugo/deps.nix]: This line seems suspect, maybe @monotux didn't update the deps with go2nix?

**Comment by peterhoeg** [comment]
Any security fixes here? We typically only back port to stable for those.

>

File: pkgs/applications/misc/hugo/deps.nix (hunk 1 above)
Comment (monotux) [reply pkgs/applications/misc/hugo/deps.nix]: I tried to merge the updates from master to this branch, but I guess I didn't do a very good job :)

Next time I attempt this I will try to _really_ understand how the tooling works.

**Comment by monotux** [comment]
@peterhoeg no, just a version bump. We should probably close this PR, I thought we did updates like this to the stable version.

**Pull request closed without merging**